### PR TITLE
Update `createapplication` command

### PR DESCRIPTION
**Pull request by vector-kerr**
Fixes #
N/A

## Description of the Change

Greetings!

This pull request updates the createapplication management command.

* The `--algorighm` parameter is added which allows specification of `""`, `"RS256"`, or `"HS256"`
* The `--skip-authorization` help text is fixed
    * It was previously incorrectly described as `"The ID of the new application"`
    * It is now described as `"If set, completely bypass the authorization form, even on the first use of the application"`

Notes:
* Documentation not updated because:
    * the command does not appear to be included in documentation
    * the command is self-documented (arguments and help strings displayed on invocation)

I hope you find this change useful!
\- Daniel.

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [x] unit-test added
- [ ] documentation updated
- [x] `CHANGELOG.md` updated (only for user relevant changes)
- [x] author name in `AUTHORS`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,9 +32,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   If you've [customized OIDC responses](https://django-oauth-toolkit.readthedocs.io/en/latest/oidc.html#customizing-the-oidc-responses)
   and want to retain the pre-2.x behavior, set `oidc_claim_scope = None` in your subclass of `OAuth2Validator`.
 * #1108 OIDC: Make the `access_token` available to `get_oidc_claims` when called from `get_userinfo_claims`.
+* Added `--algorithm` argument to `createapplication` management command
```
Comment (n2ygk) [inline CHANGELOG.md:35]:
```suggestion
* #1132: Added `--algorithm` argument to `createapplication` management command
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,9 +32,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Fixed
 * #1108 OIDC: Fix `validate_bearer_token()` to properly set `request.scopes` to the list of granted scopes.
+* Fixed help text for `--skip-authorization` argument of the `createapplication` management command
```
Comment (n2ygk) [inline CHANGELOG.md:39]:
```suggestion
* #1132: Fixed help text for `--skip-authorization` argument of the `createapplication` management command
```

**Review by n2ygk — CHANGES_REQUESTED** [review]
Hey, thanks for this!

Would you consider adding something to the [docs](https://github.com/jazzband/django-oauth-toolkit/blob/master/docs/management_commands.rst) for this command? It's never a bad thing to improve the documentation.

**Comment by vector-kerr** [comment]
@n2ygk Sure thing, I've had a go at it in commit de0db70.

I'm definitely not an expert on RST documentation (much more familiar and comfortable with markdown) but I gave it a shot.

Someone will probably want to read over it to make sure I haven't buggered it up. I checked it in VSCode with an RST plugin and it seems ok, but I can't be certain.

\- Daniel

**Comment by n2ygk** [comment]
> @n2ygk Sure thing, I've had a go at it in commit [de0db70](https://github.com/jazzband/django-oauth-toolkit/commit/de0db700a3da94b44e5239461ca4eb130b25f662).
> 
> I'm definitely not an expert on RST documentation (much more familiar and comfortable with markdown) but I gave it a shot.

I think most of us prefer markdown.
> 
> Someone will probably want to read over it to make sure I haven't buggered it up. I checked it in VSCode with an RST plugin and it seems ok, but I can't be certain.

To see what your docs edit looks like, you can locally run `tox -e docs` and then open `docs/_build/html/index.html` to see how it looks. You can similarly click on the [Details](https://django-oauth-toolkit--1132.org.readthedocs.build/en/1132/) for this check: `docs/readthedocs.org:django-oauth-toolkit — Read the Docs build succeeded!`

**Comment by n2ygk** [comment]
Weirdly the docs build failed even though it works locally.... I'll have to dig a bit further on that.

**Comment by n2ygk** [comment]
> Weirdly the docs build failed even though it works locally.... I'll have to dig a bit further on that.

This project's tox setup doesn't refresh the `.tox` cache properly. After `rm -r .tox/docs` I can reproduce the error.

**Comment by n2ygk** [comment]
> > Weirdly the docs build failed even though it works locally.... I'll have to dig a bit further on that.
> 
> This project's tox setup doesn't refresh the `.tox` cache properly. After `rm -r .tox/docs` I can reproduce the error.

Unrelated problem: https://github.com/sphinx-doc/sphinx/issues/10291

**Comment by vector-kerr** [comment]
Hello again @n2ygk,

I've just merged the master branch into my topical which I presume adds the version pin for jinja2.
If there's anything else you need from my end please let me know. :)

\- Daniel

**Comment by n2ygk** [comment]
> Hello again @n2ygk,
> 
> I've just merged the master branch into my topical which I presume adds the version pin for jinja2. If there's anything else you need from my end please let me know. :)
> 
> - Daniel

Yeah I'll make sure to resync it with master as there are a few PRs in flight right now. Please either accept or reject the documentation suggestion I made w.r.t table vs. output of `manage.py createapplication --help`.

Thanks!

**Comment by vector-kerr** [comment]
> Please either accept or reject the documentation suggestion I made w.r.t table vs. output of `manage.py createapplication --help`.

Hello again @n2ygk,

Sorry to be a pain, but I don't seem to be able to find the documentation suggestion you're referring to.

GitHub shows me a big red `"1 change requested"` alert in this conversation, but when I click through with the "View changes" button, it just takes me to the review page where there are no unresolved review requests (that I can see). I also cannot see any obvious difference to the `management_commands.rst` documentation file that you might have made in the diff.

I do use GitLab a lot more than GitHub so I may also just not be looking in the right place...

I'm happy to look over it, I just can't seem to get to it!

**Review by n2ygk — CHANGES_REQUESTED** [review]
Sorry. I keep doing this. I left my suggestion Pending.

File: docs/management_commands.rst [hunk 3]
```diff
@@ -21,3 +23,39 @@ To prevent the CPU and RAM high peaks during deletion process use ``CLEAR_EXPIRE
...
+| `--skip-authorization`     | flag | If set, completely bypass the authorization form, even on the first use of the application      |
++----------------------------+------+-------------------------------------------------------------------------------------------------+
+| `--algorithm`              | str  | The OIDC token signing algorithm for this application (e.g., `RS256` or `HS256`)                |
++----------------------------+------+-------------------------------------------------------------------------------------------------+
+| `client_type`              | str  | The client type, can be `confidential` or `public`                                              |
++----------------------------+------+-------------------------------------------------------------------------------------------------+
+| `authorization_grant_type` | str  | The type of authorization grant to be used                                                      |
++----------------------------+------+-------------------------------------------------------------------------------------------------+
```
[the remark this answers is not in the document]
Comment (n2ygk) [reply docs/management_commands.rst:61]: @vector-kerr see above. Apologies for neglecting to complete by review feedback so it showed for me and I ignored that it was pending.

**Comment by vector-kerr** [comment]
> Sorry. I keep doing this. I left my suggestion Pending.

Not a problem!
Just committed your suggestion, so I think we're good to go.
Can I please get you to run the workflows once more? (I can't do it myself as a first-time contributor.)
Thanks for your patience and help!

\- Daniel

File: docs/management_commands.rst [hunk 4]
```diff
@@ -21,3 +23,43 @@ To prevent the CPU and RAM high peaks during deletion process use ``CLEAR_EXPIRE
 
 Note: Refresh tokens need to expire before AccessTokens can be removed from the
 database. Using ``cleartokens`` without ``REFRESH_TOKEN_EXPIRE_SECONDS`` has limited effect.
+
```
Comment (n2ygk) [inline docs/management_commands.rst:26]: Ugh, I forgot this is RsT and not markdown so my suggestion using triple-backticks wasn't so helpful....

Comment (n2ygk) [reply docs/management_commands.rst:26]: I pushed a commit with the required indentation.

**Review by n2ygk — APPROVED** [review]

**Pull request merged**